File: src/main/java/org/apache/fineract/client/services/FundsApi.java
/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements. See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership. The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
package org.apache.fineract.client.services;

import org.apache.fineract.client.CollectionFormats.*;

import rx.Observable;
import retrofit2.http.*;

import okhttp3.RequestBody;
import okhttp3.ResponseBody;
import okhttp3.MultipartBody;

import org.apache.fineract.client.models.GetFundsResponse;
import org.apache.fineract.client.models.PostFundsRequest;
import org.apache.fineract.client.models.PostFundsResponse;
import org.apache.fineract.client.models.PutFundsFundIdRequest;
import org.apache.fineract.client.models.PutFundsFundIdResponse;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public interface FundsApi {
  /**
   * Create a Fund
   * Creates a Fund
   * @param postFundsRequest  (required)
   * @return Observable&lt;PostFundsResponse&gt;
   */
  @Headers({
    "Content-Type:application/json"
  })
  @POST("funds")
  Observable<PostFundsResponse> createFund(
    @retrofit2.http.Body PostFundsRequest postFundsRequest
  );

  /**
   * Retrieve a Fund
   * Returns the details of a Fund.  Example Requests:  funds/1
   * @param fundId fundId (required)
   * @return Observable&lt;GetFundsResponse&gt;
   */
  @GET("funds/{fundId}")
  Observable<GetFundsResponse> retreiveFund(
    @retrofit2.http.Path("fundId") Long fundId
  );

  /**
   * Retrieve Funds
   * Returns the list of funds.  Example Requests:  funds
   * @return Observable&lt;List&lt;GetFundsResponse&gt;&gt;
   */
  @GET("funds")
  Observable<List<GetFundsResponse>> retrieveFunds();
    

  /**
   * Update a Fund
   * Updates the details of a fund.
   * @param fundId fundId (required)
   * @param putFundsFundIdRequest  (required)
   * @return Observable&lt;PutFundsFundIdResponse&gt;
   */
  @Headers({
    "Content-Type:application/json"
  })
  @PUT("funds/{fundId}")
  Observable<PutFundsFundIdResponse> updateFund(
    @retrofit2.http.Path("fundId") Long fundId, @retrofit2.http.Body PutFundsFundIdRequest putFundsFundIdRequest
  );

}

File: src/main/kotlin/org/openapitools/client/apis/FundsApi.kt
package org.openapitools.client.apis

import org.openapitools.client.models.GetFundsResponse
import org.openapitools.client.models.PostFundsRequest
import org.openapitools.client.models.PostFundsResponse
import org.openapitools.client.models.PutFundsFundIdRequest
import org.openapitools.client.models.PutFundsFundIdResponse
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.PUT
import retrofit2.http.Path

interface FundsApi {
    /**
     * Create a Fund
     * Creates a Fund
     * Responses:
     *  - 200: OK
     *
     * @param postFundsRequest
     * @return [PostFundsResponse]
     */
    @POST("v1/funds")
    suspend fun createFund(@Body postFundsRequest: PostFundsRequest): PostFundsResponse

    /**
     * Retrieve a Fund
     * Returns the details of a Fund.  Example Requests:  funds/1
     * Responses:
     *  - 200: OK
     *
     * @param fundId fundId
     * @return [GetFundsResponse]
     */
    @GET("v1/funds/{fundId}")
    suspend fun retrieveFund(@Path("fundId") fundId: Long): GetFundsResponse

    /**
     * Retrieve Funds
     * Returns the list of funds.  Example Requests:  funds
     * Responses:
     *  - 200: OK
     *
     * @return [kotlin.collections.List<GetFundsResponse]
     */
    @GET("v1/funds")
    suspend fun retrieveFunds(): List<GetFundsResponse>

    /**
     * Update a Fund
     * Updates the details of a fund.
     * Responses:
     *  - 200: OK
     *
     * @param fundId fundId
     * @param putFundsFundIdRequest
     * @return [PutFundsFundIdResponse]
     */
    @PUT("v1/funds/{fundId}")
    suspend fun updateFund(
        @Path("fundId") fundId: Long,
        @Body putFundsFundIdRequest: PutFundsFundIdRequest
    ): PutFundsFundIdResponse

}
